src/Perl6/Optimizer.nqp

```diff
@@ -260,28 +260,47 @@ my class Symbols {
         return 0;
     }
 
+    # method find_in_setting($symbol) {
+    #     if !nqp::defined($!SETTING) {
+    #         my int $i := +@!block_stack;
+    #         while $i > 0 && !nqp::defined($!SETTING) {
+    #             $i := $i - 1;
+    #             my $block := @!block_stack[$i];
+    #             my %sym := $block.symbol("!CORE_MARKER");
+    #             if +%sym {
+    #                 $!SETTING := $block;
+    #             }
+    #         }
+    #         if !nqp::defined($!SETTING) {
+    #             nqp::die("Optimizer couldn't find CORE while looking for $symbol.");
+    #         }
+    #     } else {
+    #         if nqp::existskey(%!SETTING_CACHE, $symbol) {
+    #             return %!SETTING_CACHE{$symbol};
+    #         }
+    #     }
+    #     my %sym := $!SETTING.symbol($symbol);
+    #     if +%sym {
+    #         return %!SETTING_CACHE{$symbol} := self.force_value(%sym, $symbol, 1);
+    #     }
+    #     nqp::die("Optimizer couldn't find $symbol in SETTING.");
+    # }
+
     method find_in_setting($symbol) {
-        if !nqp::defined($!SETTING) {
-            my int $i := +@!block_stack;
-            while $i > 0 && !nqp::defined($!SETTING) {
-                $i := $i - 1;
-                my $block := @!block_stack[$i];
-                my %sym := $block.symbol("!CORE_MARKER");
-                if +%sym {
-                    $!SETTING := $block;
-                }
-            }
-            if !nqp::defined($!SETTING) {
-                nqp::die("Optimizer couldn't find CORE while looking for $symbol.");
-            }
-        } else {
-            if nqp::existskey(%!SETTING_CACHE, $symbol) {
-                return %!SETTING_CACHE{$symbol};
-            }
+        if nqp::existskey(%!SETTING_CACHE, $symbol) {
```

Comment:
Am I seeing the same method twice ??

Reply:
The previous version is commented out. Mine is still active. I wasn't sure if my version is not flawed and temporarily left the old one in the comments.

Thanks, Liz!

BTW, this all is coming from attempt to make `QuantHash`es iterables in 6.e.